Index: configure.ac
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/configure.ac,v
retrieving revision 1.146
diff -u -r1.146 configure.ac
--- configure.ac	2 Feb 2009 05:35:52 -0000	1.146
+++ configure.ac	25 Feb 2009 06:03:31 -0000
@@ -458,6 +458,14 @@
 fi
 AM_CONDITIONAL(ENABLE_JPEG_SERVER, test "x$ac_cv_header_simgear_screen_jpgfactory_hxx" = "xyes")
 
+dnl Check for installed OpenFDM
+AC_CHECK_HEADER(OpenFDM/OpenFDMConfig.h)
+AM_CONDITIONAL(ENABLE_OpenFDM_FDM, \
+               [ test "x$ac_cv_header_OpenFDM_OpenFDMConfig_h" = "xyes" ] )
+if test "x$ac_cv_header_OpenFDM_OpenFDMConfig_h" = "xyes" ; then
+    AC_DEFINE([FG_ENABLE_OPENFDM_FDM], 1, [Define for no logging output])
+fi
+
 # Check for "plib" without which we cannot go on
 AC_CHECK_HEADER(plib/ul.h)
 if test "x$ac_cv_header_plib_ul_h" != "xyes"; then
@@ -688,6 +696,7 @@
 	src/FDM/JSBSim/models/atmosphere/Makefile \
 	src/FDM/JSBSim/models/propulsion/Makefile \
 	src/FDM/LaRCsim/Makefile \
+	src/FDM/OpenFDM/Makefile \
 	src/FDM/SP/Makefile \
 	src/FDM/UIUCModel/Makefile \
 	src/FDM/YASim/Makefile \
Index: src/FDM/Makefile.am
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/FDM/Makefile.am,v
retrieving revision 1.8
diff -u -r1.8 Makefile.am
--- src/FDM/Makefile.am	25 Jul 2008 18:38:30 -0000	1.8
+++ src/FDM/Makefile.am	25 Feb 2009 06:03:32 -0000
@@ -4,8 +4,14 @@
 SP_DIR =
 endif
 
+if ENABLE_OpenFDM_FDM
+OpenFDM_DIR = OpenFDM
+else
+OpenFDM_DIR =
+endif
+
 SUBDIRS	= JSBSim LaRCsim UIUCModel YASim \
-          $(SP_DIR) ExternalNet ExternalPipe
+          $(SP_DIR) $(OpenFDM_DIR) ExternalNet ExternalPipe
 
 noinst_LIBRARIES = libFlight.a
 
Index: src/FDM/flight.hxx
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/FDM/flight.hxx,v
retrieving revision 1.18
diff -u -r1.18 flight.hxx
--- src/FDM/flight.hxx	27 Jul 2008 16:25:14 -0000	1.18
+++ src/FDM/flight.hxx	25 Feb 2009 06:03:32 -0000
@@ -347,7 +347,10 @@
 	FG_PARACHUTE = 9,
 
 	// Driven externally via a serial port, net, file, etc.
-	FG_EXTERNAL = 10
+	FG_EXTERNAL = 10,
+
+	// OpenFDM. Reads OpenFDM and JSBSim files.
+	FG_OPENFDM = 11
     };
 
     // initialization
Index: src/Main/Makefile.am
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/Main/Makefile.am,v
retrieving revision 1.88
diff -u -r1.88 Makefile.am
--- src/Main/Makefile.am	13 Feb 2009 11:52:49 -0000	1.88
+++ src/Main/Makefile.am	25 Feb 2009 06:03:33 -0000
@@ -9,6 +9,13 @@
 SP_FDM_LIBS = 
 endif
 
+if ENABLE_OpenFDM_FDM
+OpenFDM_LIBS = $(top_builddir)/src/FDM/OpenFDM/libFGOpenFDM.a \
+               -lOpenFDMJSBReader -lOpenFDMeasyxmlXML -lOpenFDMXML -lOpenFDM -lrt
+else
+OpenFDM_LIBS = 
+endif
+
 if WITH_THREADS
 THREAD_LIBS = -lsgthreads $(thread_LIBS)
 else
@@ -74,6 +81,7 @@
 	$(top_builddir)/src/FDM/LaRCsim/libLaRCsim.a \
 	$(top_builddir)/src/FDM/UIUCModel/libUIUCModel.a \
 	$(SP_FDM_LIBS) \
+	$(OpenFDM_LIBS) \
 	$(top_builddir)/src/GUI/libGUI.a \
 	$(top_builddir)/src/Autopilot/libAutopilot.a \
 	$(top_builddir)/src/Input/libInput.a \
Index: src/Main/fg_init.cxx
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/Main/fg_init.cxx,v
retrieving revision 1.227
diff -u -r1.227 fg_init.cxx
--- src/Main/fg_init.cxx	10 Jan 2009 16:48:22 -0000	1.227
+++ src/Main/fg_init.cxx	25 Feb 2009 06:03:34 -0000
@@ -91,6 +91,9 @@
 #include <FDM/UFO.hxx>
 #include <FDM/NullFDM.hxx>
 #include <FDM/YASim/YASim.hxx>
+#ifdef FG_ENABLE_OPENFDM_FDM
+#include <FDM/OpenFDM/FGOpenFDM.h>
+#endif
 #include <GUI/new_gui.hxx>
 #include <Include/general.hxx>
 #include <Input/input.hxx>
@@ -1279,6 +1282,10 @@
         cur_fdm_state = new FGNullFDM( dt );
     } else if ( model == "yasim" ) {
         cur_fdm_state = new YASim( dt );
+#ifdef FG_ENABLE_OPENFDM_FDM
+   } else if ( model == "openfdm" ) {
+        cur_fdm_state = new OpenFDM::FGOpenFDM();
+#endif
     } else {
         throw sg_throwable(string("Unrecognized flight model '") + model
                + "', cannot init flight dynamics model.");
