diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1ec1f62..a141ae7 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -54,6 +54,7 @@ option(ENABLE_LARCSIM "Set to ON to build FlightGear with LaRCsim FDM" ON)
 option(ENABLE_YASIM "Set to ON to build FlightGear with YASIM FDM" ON)
 option(ENABLE_JSBSIM "Set to ON to build FlightGear with JSBSim FDM" ON)
 option(ENABLE_FGADMIN "Set to ON to build FlightGear with FGADMIN" ON)
+option(FG_ENABLE_OPENFDM_FDM "Set to ON to build FlightGear with OpenFDM FDM" ON)
 option(EVENT_INPUT "Set to ON to build FlightGear with event-based Input support" OFF)
 option(ENABLE_LIBSVN "Set to ON to build terrasync with libsvnclient support" OFF)
 
diff --git a/configure.ac b/configure.ac
index 3635cfe..cb0ff54 100644
--- a/configure.ac
+++ b/configure.ac
@@ -646,6 +646,14 @@ if test "x$ac_cv_header_simgear_screen_jpgfactory_hxx" = "xyes"; then
 fi
 AM_CONDITIONAL(ENABLE_JPEG_SERVER, test "x$ac_cv_header_simgear_screen_jpgfactory_hxx" = "xyes")
 
+dnl Check for installed OpenFDM
+AC_CHECK_HEADER(OpenFDM/OpenFDMConfig.h)
+AM_CONDITIONAL(ENABLE_OpenFDM_FDM, \
+               [ test "x$ac_cv_header_OpenFDM_OpenFDMConfig_h" = "xyes" ] )
+if test "x$ac_cv_header_OpenFDM_OpenFDMConfig_h" = "xyes" ; then
+    AC_DEFINE([FG_ENABLE_OPENFDM_FDM], 1, [Define for no logging output])
+fi
+
 # Check for "plib" without which we cannot go on
 case "${host}" in
 *-apple-darwin*)
@@ -914,6 +922,7 @@ AC_CONFIG_FILES([ \
 	src/FDM/JSBSim/models/atmosphere/Makefile \
 	src/FDM/JSBSim/models/propulsion/Makefile \
 	src/FDM/LaRCsim/Makefile \
+	src/FDM/OpenFDM/Makefile \
 	src/FDM/SP/Makefile \
 	src/FDM/UIUCModel/Makefile \
 	src/FDM/YASim/Makefile \
diff --git a/src/FDM/CMakeLists.txt b/src/FDM/CMakeLists.txt
index 74d695f..70ca3d6 100644
--- a/src/FDM/CMakeLists.txt
+++ b/src/FDM/CMakeLists.txt
@@ -9,6 +9,12 @@ set(SP_FDM_SOURCES
 	)
 endif()
 
+if(FG_ENABLE_OPENFDM_FDM)
+set(OPENFDM_SOURCES
+	OpenFDM/FGOpenFDM.cpp
+	)
+endif()
+
 set(UIUC_SOURCES
 	uiuc_1DdataFileReader.cpp
 	uiuc_1Dinterpolation.cpp
@@ -133,6 +139,7 @@ set(SOURCES
 	TankProperties.cxx
 	groundcache.cxx
 	${SP_FDM_SOURCES}
+	${OPENFDM_SOURCES}
 	ExternalNet/ExternalNet.cxx
 	ExternalPipe/ExternalPipe.cxx
 	)
diff --git a/src/FDM/Makefile.am b/src/FDM/Makefile.am
index 41bfe3f..b94a8d4 100644
--- a/src/FDM/Makefile.am
+++ b/src/FDM/Makefile.am
@@ -4,8 +4,14 @@ else
 SP_DIR =
 endif
 
+if ENABLE_OpenFDM_FDM
+OpenFDM_DIR = OpenFDM
+else
+OpenFDM_DIR =
+endif
+
 SUBDIRS	= JSBSim LaRCsim UIUCModel YASim \
-          $(SP_DIR) ExternalNet ExternalPipe
+          $(SP_DIR) $(OpenFDM_DIR) ExternalNet ExternalPipe
 
 noinst_LIBRARIES = libFlight.a
 
diff --git a/src/FDM/fdm_shell.cxx b/src/FDM/fdm_shell.cxx
index 8055bce..e77be1c 100644
--- a/src/FDM/fdm_shell.cxx
+++ b/src/FDM/fdm_shell.cxx
@@ -48,6 +48,9 @@
 #include <FDM/UFO.hxx>
 #include <FDM/NullFDM.hxx>
 #include <FDM/YASim/YASim.hxx>
+#ifdef FG_ENABLE_OPENFDM_FDM
+#include <FDM/OpenFDM/FGOpenFDM.h>
+#endif
 
 
 /*
@@ -245,6 +248,10 @@ void FDMShell::createImplementation()
         _impl = new FGNullFDM( dt );
     } else if ( model == "yasim" ) {
         _impl = new YASim( dt );
+#ifdef FG_ENABLE_OPENFDM_FDM
+    } else if ( model == "openfdm" ) {
+        _impl = new OpenFDM::FGOpenFDM();
+#endif
     } else {
         throw sg_exception(string("Unrecognized flight model '") + model
                + "', cannot init flight dynamics model.");
diff --git a/src/FDM/flight.hxx b/src/FDM/flight.hxx
index 6965d30..49de534 100644
--- a/src/FDM/flight.hxx
+++ b/src/FDM/flight.hxx
@@ -402,7 +402,10 @@ public:
 	FG_PARACHUTE = 9,
 
 	// Driven externally via a serial port, net, file, etc.
-	FG_EXTERNAL = 10
+	FG_EXTERNAL = 10,
+
+	// OpenFDM. Reads OpenFDM and JSBSim files.
+	FG_OPENFDM = 11
     };
 
     // initialization
diff --git a/src/Include/config_cmake.h.in b/src/Include/config_cmake.h.in
index 0477170..86f0ce4 100644
--- a/src/Include/config_cmake.h.in
+++ b/src/Include/config_cmake.h.in
@@ -28,6 +28,7 @@
 #cmakedefine ENABLE_LARCSIM
 #cmakedefine ENABLE_YASIM
 #cmakedefine ENABLE_JSBSIM
+#cmakedefine FG_ENABLE_OPENFDM_FDM
 
 #cmakedefine HAVE_SVN_CLIENT_H
 #cmakedefine HAVE_LIBSVN_CLIENT_1
diff --git a/src/Main/CMakeLists.txt b/src/Main/CMakeLists.txt
index 954c020..7fd0dd8 100644
--- a/src/Main/CMakeLists.txt
+++ b/src/Main/CMakeLists.txt
@@ -47,9 +47,16 @@ else()
   set(HLA_LIBRARIES "")
 endif()
 
+if(FG_ENABLE_OPENFDM_FDM)
+  set(OPENFDM_LIBRARIES "-lOpenFDMJSBReader -lOpenFDMeasyxmlXML -lOpenFDMXML -lOpenFDM")
+else()
+  set(OPENFDM_LIBRARIES "")
+endif()
+
 target_link_libraries(fgfs 
 	${FG_LIBS}
         ${HLA_LIBRARIES}
+	${OPENFDM_LIBRARIES}
 	${SIMGEAR_LIBRARIES}
 	${OPENSCENEGRAPH_LIBRARIES} 
 	${OPENAL_LIBRARY} 
diff --git a/src/Main/Makefile.am b/src/Main/Makefile.am
index c4525ed..6abb582 100644
--- a/src/Main/Makefile.am
+++ b/src/Main/Makefile.am
@@ -9,6 +9,13 @@ else
 SP_FDM_LIBS =
 endif
 
+if ENABLE_OpenFDM_FDM
+OpenFDM_LIBS = $(top_builddir)/src/FDM/OpenFDM/libFGOpenFDM.a \
+               -lOpenFDMJSBReader -lOpenFDMeasyxmlXML -lOpenFDMXML -lOpenFDM -lrt
+else
+OpenFDM_LIBS = 
+endif
+
 if WITH_EVENTINPUT
 EVENT_LIBS = $(eventinput_LIBS)
 else
@@ -90,6 +97,7 @@ fgfs_LDADD = \
 	$(top_builddir)/src/FDM/LaRCsim/libLaRCsim.a \
 	$(top_builddir)/src/FDM/UIUCModel/libUIUCModel.a \
 	$(SP_FDM_LIBS) \
+	$(OpenFDM_LIBS) \
 	$(top_builddir)/src/GUI/libGUI.a \
 	$(top_builddir)/src/Autopilot/libAutopilot.a \
 	$(top_builddir)/src/Input/libInput.a \
