Index: configure.ac
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/configure.ac,v
retrieving revision 1.161
diff -u -r1.161 configure.ac
--- configure.ac	30 Dec 2009 14:11:16 -0000	1.161
+++ configure.ac	30 Dec 2009 18:47:56 -0000
@@ -550,6 +550,14 @@
 fi
 AM_CONDITIONAL(ENABLE_JPEG_SERVER, test "x$ac_cv_header_simgear_screen_jpgfactory_hxx" = "xyes")
 
+dnl Check for installed OpenFDM
+AC_CHECK_HEADER(OpenFDM/OpenFDMConfig.h)
+AM_CONDITIONAL(ENABLE_OpenFDM_FDM, \
+               [ test "x$ac_cv_header_OpenFDM_OpenFDMConfig_h" = "xyes" ] )
+if test "x$ac_cv_header_OpenFDM_OpenFDMConfig_h" = "xyes" ; then
+    AC_DEFINE([FG_ENABLE_OPENFDM_FDM], 1, [Define for no logging output])
+fi
+
 # Check for "plib" without which we cannot go on
 case "${host}" in
 *-apple-darwin*)
@@ -816,6 +824,7 @@
 	src/FDM/JSBSim/models/atmosphere/Makefile \
 	src/FDM/JSBSim/models/propulsion/Makefile \
 	src/FDM/LaRCsim/Makefile \
+	src/FDM/OpenFDM/Makefile \
 	src/FDM/SP/Makefile \
 	src/FDM/UIUCModel/Makefile \
 	src/FDM/YASim/Makefile \
Index: src/FDM/Makefile.am
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/FDM/Makefile.am,v
retrieving revision 1.8
diff -u -r1.8 Makefile.am
--- src/FDM/Makefile.am	25 Jul 2008 18:38:30 -0000	1.8
+++ src/FDM/Makefile.am	30 Dec 2009 18:47:56 -0000
@@ -4,8 +4,14 @@
 SP_DIR =
 endif
 
+if ENABLE_OpenFDM_FDM
+OpenFDM_DIR = OpenFDM
+else
+OpenFDM_DIR =
+endif
+
 SUBDIRS	= JSBSim LaRCsim UIUCModel YASim \
-          $(SP_DIR) ExternalNet ExternalPipe
+          $(SP_DIR) $(OpenFDM_DIR) ExternalNet ExternalPipe
 
 noinst_LIBRARIES = libFlight.a
 
Index: src/FDM/flight.hxx
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/FDM/flight.hxx,v
retrieving revision 1.22
diff -u -r1.22 flight.hxx
--- src/FDM/flight.hxx	17 Mar 2009 06:51:52 -0000	1.22
+++ src/FDM/flight.hxx	30 Dec 2009 18:47:56 -0000
@@ -345,7 +345,10 @@
 	FG_PARACHUTE = 9,
 
 	// Driven externally via a serial port, net, file, etc.
-	FG_EXTERNAL = 10
+	FG_EXTERNAL = 10,
+
+	// OpenFDM. Reads OpenFDM and JSBSim files.
+	FG_OPENFDM = 11
     };
 
     // initialization
Index: src/Main/Makefile.am
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/Main/Makefile.am,v
retrieving revision 1.94
diff -u -r1.94 Makefile.am
--- src/Main/Makefile.am	30 Dec 2009 14:11:38 -0000	1.94
+++ src/Main/Makefile.am	30 Dec 2009 18:47:57 -0000
@@ -9,6 +9,13 @@
 SP_FDM_LIBS = 
 endif
 
+if ENABLE_OpenFDM_FDM
+OpenFDM_LIBS = $(top_builddir)/src/FDM/OpenFDM/libFGOpenFDM.a \
+               -lOpenFDMJSBReader -lOpenFDMeasyxmlXML -lOpenFDMXML -lOpenFDM -lrt
+else
+OpenFDM_LIBS = 
+endif
+
 if ENABLE_ATCDCL
 ATCDCL_LIBS = $(top_builddir)/src/ATCDCL/libATCDCL.a
 else
@@ -100,6 +107,7 @@
 	$(top_builddir)/src/FDM/LaRCsim/libLaRCsim.a \
 	$(top_builddir)/src/FDM/UIUCModel/libUIUCModel.a \
 	$(SP_FDM_LIBS) \
+	$(OpenFDM_LIBS) \
 	$(top_builddir)/src/GUI/libGUI.a \
 	$(top_builddir)/src/Autopilot/libAutopilot.a \
 	$(top_builddir)/src/Input/libInput.a \
Index: src/Main/fg_init.cxx
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/Main/fg_init.cxx,v
retrieving revision 1.246
diff -u -r1.246 fg_init.cxx
--- src/Main/fg_init.cxx	30 Dec 2009 14:11:41 -0000	1.246
+++ src/Main/fg_init.cxx	30 Dec 2009 18:47:58 -0000
@@ -101,6 +101,9 @@
 #include <FDM/UFO.hxx>
 #include <FDM/NullFDM.hxx>
 #include <FDM/YASim/YASim.hxx>
+#ifdef FG_ENABLE_OPENFDM_FDM
+#include <FDM/OpenFDM/FGOpenFDM.h>
+#endif
 #include <GUI/new_gui.hxx>
 #include <Include/general.hxx>
 #include <Input/input.hxx>
@@ -1298,6 +1301,10 @@
         cur_fdm_state = new FGNullFDM( dt );
     } else if ( model == "yasim" ) {
         cur_fdm_state = new YASim( dt );
+#ifdef FG_ENABLE_OPENFDM_FDM
+   } else if ( model == "openfdm" ) {
+        cur_fdm_state = new OpenFDM::FGOpenFDM();
+#endif
     } else {
         throw sg_exception(string("Unrecognized flight model '") + model
                + "', cannot init flight dynamics model.");
