diff -urNb flightgear/configure.ac flightgear-OpenFDM/configure.ac
--- flightgear/configure.ac	2011-02-04 21:40:50.243193039 +0100
+++ flightgear-OpenFDM/configure.ac	2011-02-04 21:43:59.514302275 +0100
@@ -638,6 +638,14 @@
 fi
 AM_CONDITIONAL(ENABLE_JPEG_SERVER, test "x$ac_cv_header_simgear_screen_jpgfactory_hxx" = "xyes")
 
+dnl Check for installed OpenFDM
+AC_CHECK_HEADER(OpenFDM/OpenFDMConfig.h)
+AM_CONDITIONAL(ENABLE_OpenFDM_FDM, \
+               [ test "x$ac_cv_header_OpenFDM_OpenFDMConfig_h" = "xyes" ] )
+if test "x$ac_cv_header_OpenFDM_OpenFDMConfig_h" = "xyes" ; then
+    AC_DEFINE([FG_ENABLE_OPENFDM_FDM], 1, [Define for no logging output])
+fi
+
 # Check for "plib" without which we cannot go on
 case "${host}" in
 *-apple-darwin*)
@@ -906,6 +914,7 @@
 	src/FDM/JSBSim/models/atmosphere/Makefile \
 	src/FDM/JSBSim/models/propulsion/Makefile \
 	src/FDM/LaRCsim/Makefile \
+	src/FDM/OpenFDM/Makefile \
 	src/FDM/SP/Makefile \
 	src/FDM/UIUCModel/Makefile \
 	src/FDM/YASim/Makefile \
diff -urNb flightgear/src/FDM/fdm_shell.cxx flightgear-OpenFDM/src/FDM/fdm_shell.cxx
--- flightgear/src/FDM/fdm_shell.cxx	2011-02-04 21:40:51.239883094 +0100
+++ flightgear-OpenFDM/src/FDM/fdm_shell.cxx	2011-02-04 21:46:38.588038239 +0100
@@ -48,6 +48,9 @@
 #include <FDM/UFO.hxx>
 #include <FDM/NullFDM.hxx>
 #include <FDM/YASim/YASim.hxx>
+#ifdef FG_ENABLE_OPENFDM_FDM
+#include <FDM/OpenFDM/FGOpenFDM.h>
+#endif
 
 /*
  * Evil global variable required by Network/FGNative,
@@ -258,6 +261,10 @@
         _impl = new FGNullFDM( dt );
     } else if ( model == "yasim" ) {
         _impl = new YASim( dt );
+#ifdef FG_ENABLE_OPENFDM_FDM
+    } else if ( model == "openfdm" ) {
+        _impl = new OpenFDM::FGOpenFDM();
+#endif
     } else {
         throw sg_exception(string("Unrecognized flight model '") + model
                + "', cannot init flight dynamics model.");
diff -urNb flightgear/src/FDM/flight.hxx flightgear-OpenFDM/src/FDM/flight.hxx
--- flightgear/src/FDM/flight.hxx	2011-02-04 21:40:51.239883094 +0100
+++ flightgear-OpenFDM/src/FDM/flight.hxx	2011-02-04 21:43:59.514302275 +0100
@@ -402,7 +402,10 @@
 	FG_PARACHUTE = 9,
 
 	// Driven externally via a serial port, net, file, etc.
-	FG_EXTERNAL = 10
+	FG_EXTERNAL = 10,
+
+	// OpenFDM. Reads OpenFDM and JSBSim files.
+	FG_OPENFDM = 11
     };
 
     // initialization
diff -urNb flightgear/src/FDM/Makefile.am flightgear-OpenFDM/src/FDM/Makefile.am
--- flightgear/src/FDM/Makefile.am	2011-02-04 21:40:50.976543581 +0100
+++ flightgear-OpenFDM/src/FDM/Makefile.am	2011-02-04 21:43:59.514302275 +0100
@@ -4,8 +4,14 @@
 SP_DIR =
 endif
 
+if ENABLE_OpenFDM_FDM
+OpenFDM_DIR = OpenFDM
+else
+OpenFDM_DIR =
+endif
+
 SUBDIRS	= JSBSim LaRCsim UIUCModel YASim \
-          $(SP_DIR) ExternalNet ExternalPipe
+          $(SP_DIR) $(OpenFDM_DIR) ExternalNet ExternalPipe
 
 noinst_LIBRARIES = libFlight.a
 
diff -urNb flightgear/src/Main/Makefile.am flightgear-OpenFDM/src/Main/Makefile.am
--- flightgear/src/Main/Makefile.am	2011-02-04 21:40:51.856564229 +0100
+++ flightgear-OpenFDM/src/Main/Makefile.am	2011-02-04 21:44:34.298452374 +0100
@@ -9,6 +9,13 @@
 SP_FDM_LIBS =
 endif
 
+if ENABLE_OpenFDM_FDM
+OpenFDM_LIBS = $(top_builddir)/src/FDM/OpenFDM/libFGOpenFDM.a \
+               -lOpenFDMJSBReader -lOpenFDMeasyxmlXML -lOpenFDMXML -lOpenFDM -lrt
+else
+OpenFDM_LIBS = 
+endif
+
 if WITH_EVENTINPUT
 EVENT_LIBS = $(eventinput_LIBS)
 else
@@ -90,6 +97,7 @@
 	$(top_builddir)/src/FDM/LaRCsim/libLaRCsim.a \
 	$(top_builddir)/src/FDM/UIUCModel/libUIUCModel.a \
 	$(SP_FDM_LIBS) \
+	$(OpenFDM_LIBS) \
 	$(top_builddir)/src/GUI/libGUI.a \
 	$(top_builddir)/src/Autopilot/libAutopilot.a \
 	$(top_builddir)/src/Input/libInput.a \
