Index: configure.ac
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/configure.ac,v
retrieving revision 1.110
diff -u -r1.110 configure.ac
--- configure.ac	21 Jan 2006 10:06:09 -0000	1.110
+++ configure.ac	5 Feb 2006 12:39:39 -0000
@@ -459,6 +468,14 @@
 fi
 AM_CONDITIONAL(ENABLE_JPEG_SERVER, test "x$ac_cv_header_simgear_screen_jpgfactory_hxx" = "xyes")
 
+dnl Check for installed OpenFDM
+AC_CHECK_HEADER(OpenFDM/OpenFDMConfig.h)
+AM_CONDITIONAL(ENABLE_OpenFDM_FDM, \
+               [ test "x$ac_cv_header_OpenFDM_OpenFDMConfig_h" = "xyes" ] )
+if test "x$ac_cv_header_OpenFDM_OpenFDMConfig_h" = "xyes" ; then
+    AC_DEFINE([FG_ENABLE_OPENFDM_FDM], 1, [Define for no logging output])
+fi
+
 AC_LANG_POP
 
 dnl Check for system installed zlib
@@ -539,6 +556,7 @@
 	src/FDM/JSBSim/models/atmosphere/Makefile \
 	src/FDM/JSBSim/models/propulsion/Makefile \
 	src/FDM/LaRCsim/Makefile \
+	src/FDM/OpenFDM/Makefile \
 	src/FDM/SP/Makefile \
 	src/FDM/UIUCModel/Makefile \
 	src/FDM/YASim/Makefile \
Index: src/FDM/Makefile.am
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/FDM/Makefile.am,v
retrieving revision 1.7
diff -u -r1.7 Makefile.am
--- src/FDM/Makefile.am	22 Nov 2004 10:10:33 -0000	1.7
+++ src/FDM/Makefile.am	5 Feb 2006 12:39:41 -0000
@@ -4,8 +4,14 @@
 SP_DIR =
 endif
 
+if ENABLE_OpenFDM_FDM
+OpenFDM_DIR = OpenFDM
+else
+OpenFDM_DIR =
+endif
+
 SUBDIRS	= Balloon JSBSim LaRCsim UIUCModel YASim \
-          $(SP_DIR) ExternalNet ExternalPipe
+          $(SP_DIR) $(OpenFDM_DIR) ExternalNet ExternalPipe
 
 noinst_LIBRARIES = libFlight.a
 
Index: src/FDM/flight.hxx
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/FDM/flight.hxx,v
retrieving revision 1.11
diff -u -r1.11 flight.hxx
--- src/FDM/flight.hxx	20 Jan 2006 17:19:02 -0000	1.11
+++ src/FDM/flight.hxx	5 Feb 2006 12:39:42 -0000
@@ -433,7 +433,10 @@
 	FG_PARACHUTE = 9,
 
 	// Driven externally via a serial port, net, file, etc.
-	FG_EXTERNAL = 10
+	FG_EXTERNAL = 10,
+
+	// OpenFDM. Reads JSBSim legacy and new files.
+	FG_OPENFDM = 11
     };
 
     // initialization
Index: src/Main/Makefile.am
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/Main/Makefile.am,v
retrieving revision 1.63
diff -u -r1.63 Makefile.am
--- src/Main/Makefile.am	20 Jan 2006 17:19:02 -0000	1.63
+++ src/Main/Makefile.am	5 Feb 2006 12:39:43 -0000
@@ -13,6 +13,13 @@
 SP_FDM_LIBS = 
 endif
 
+if ENABLE_OpenFDM_FDM
+OpenFDM_LIBS = $(top_builddir)/src/FDM/OpenFDM/libFGOpenFDM.a \
+               -lOpenFDMJSBReader -lOpenFDMeasyxmlXML -lOpenFDMXML -lOpenFDM
+else
+OpenFDM_LIBS = 
+endif
+
 if WITH_THREADS
 THREAD_LIBS = -lsgthreads $(thread_LIBS)
 else
@@ -84,6 +91,7 @@
 	$(top_builddir)/src/FDM/LaRCsim/libLaRCsim.a \
 	$(top_builddir)/src/FDM/UIUCModel/libUIUCModel.a \
 	$(SP_FDM_LIBS) \
+	$(OpenFDM_LIBS) \
 	$(top_builddir)/src/GUI/libGUI.a \
 	$(top_builddir)/src/Autopilot/libAutopilot.a \
 	$(top_builddir)/src/Input/libInput.a \
Index: src/Main/fg_init.cxx
===================================================================
RCS file: /var/cvs/FlightGear-0.9/source/src/Main/fg_init.cxx,v
retrieving revision 1.150
diff -u -r1.150 fg_init.cxx
--- src/Main/fg_init.cxx	30 Jan 2006 13:29:50 -0000	1.150
+++ src/Main/fg_init.cxx	5 Feb 2006 12:39:44 -0000
@@ -90,6 +90,9 @@
 #include <FDM/UFO.hxx>
 #include <FDM/NullFDM.hxx>
 #include <FDM/YASim/YASim.hxx>
+#ifdef FG_ENABLE_OPENFDM_FDM
+#include <FDM/OpenFDM/FGOpenFDM.h>
+#endif
 #include <GUI/new_gui.hxx>
 #include <Include/general.hxx>
 #include <Input/input.hxx>
@@ -1361,6 +1364,10 @@
             cur_fdm_state = new FGNullFDM( dt );
         } else if ( model == "yasim" ) {
             cur_fdm_state = new YASim( dt );
+#ifdef FG_ENABLE_OPENFDM_FDM
+        } else if ( model == "openfdm" ) {
+            cur_fdm_state = new OpenFDM::FGOpenFDM();
+#endif
         } else {
             SG_LOG(SG_GENERAL, SG_ALERT,
                    "Unrecognized flight model '" << model
